setup.py

```diff
@@ -13,24 +13,52 @@
 with open(os.path.join(BASE_DIR, "deflate.c"), "r") as src:
     version = re.match(r'.*#define MODULE_VERSION "(.*?)"', src.read(), re.S).group(1)
 
-cpu = "arm" if platform.machine().startswith("arm") else "x86"
+prefer_system_libdeflate = not bool(os.environ.get("USE_BUNDLED_DEFLATE"))
+system_prefix_libdeflate = os.environ.get("LIBDEFLATE_PREFIX")
+
+try:
+    import pkgconfig as pc
+except ImportError:
+    print("Warning: can not import pkgconfig python package.")
+    pc = None
+
+deflate_sources = [
+    "deflate.c",
+]
+
+if prefer_system_libdeflate:
+    if system_prefix_libdeflate:
+        print("Detected and preferring libdeflate [via LIBDEFLATE_PREFIX]")
+        extension_kwargs = dict(include_dirs=[os.path.join(system_prefix_libdeflate, "include")],
+                                library_dirs=[os.path.join(system_prefix_libdeflate, "lib")],
+                                libraries=["deflate"])
+
+    elif pc and pc.installed("libdeflate", ">= 1.10"):
+        print("Detected and preferring libdeflate [via pkgconfig]")
+        extension_kwargs = pc.parse("libdeflate")
```

Comment:
this is the block executed by default.

there's a problem now if there is no pkgconfig (pc is None) and the user did not set LIBDEFLATE_PREFIX (which is also usually the default).

so, people who do not read the docs and install the lib and pkgconfig before this, will get a build failure - **not a fallback to bundled code**.

Reply:
Good catch, should be resolved now.